apple/Sources/Sargon/Extensions/Methods/Addresses+Wrap+Functions.swift

```diff
@@ -0,0 +1,662 @@
+		public struct Preview {
+			fileprivate init() {}
+			public static let of = Self()
+
+			public let first: PackageAddress = newPackageAddressPlaceholder()
+			public let second: PackageAddress = newPackageAddressPlaceholderOther()
+		}
+
+		/// Preview values for `PackageAddress`, e.g.:
+		/// `PackageAddress.preview.first`
+		/// or
+		/// `PackageAddress.preview.second`
+		public static let preview = Preview.of
+	}
+#endif
+
+#if DEBUG
+	extension PackageAddress: CaseIterable {
+		public typealias AllCases = [Self]
+		public static var allCases: AllCases {
+			let of = Preview.of
+			return [
+				of.first,
+				of.second,
+			]
+		}
+	}
+#endif
+
+
+extension ResourceAddress: AddressProtocol {
+	public init(validatingAddress bech32String: String) throws {
+		self = try newResourceAddress(bech32: bech32String)
+	}
+
+	/// The bech32 encoded string for this address.
+	public var address: String {
+		resourceAddressBech32Address(address: self)
+	}
+
+	public var networkID: NetworkId {
+		resourceAddressNetworkId(address: self)
+	}
+
+	/// If this is an address of a **fungible** resource or not.
+	public var isFungible: Bool {
+		resourceAddressIsFungible(address: self)
+	}
+
+	/// If this is an address of a **non-fungible** resource or not.
+	public var isNonFungible: Bool {
+		resourceAddressIsNonFungible(address: self)
+	}
+}
+
+#if DEBUG
+	extension ResourceAddress {
+		/// Namespace for preview values of `ResourceAddress`
+		public struct Preview {
+			fileprivate init() {}
+			public static let of = Self()
+
+			public let mainnetXRD: ResourceAddress =
+				newResourceAddressPlaceholderMainnetXrd()
+			public let mainnetCandy: ResourceAddress =
+				newResourceAddressPlaceholderMainnetCandy()
+			/// Gumball Club membership NFT resource address
+			public let mainnetNonFungbleGCMembership: ResourceAddress =
+				newResourceAddressPlaceholderMainnetNftGcMembership()
+
+			public let stokenetXRD: ResourceAddress =
+				newResourceAddressPlaceholderStokenetXrd()
+			public let stokenetGum: ResourceAddress =
+				newResourceAddressPlaceholderStokenetGum()
+			public let stokenetGC: ResourceAddress =
+				newResourceAddressPlaceholderStokenetGcTokens()
+			public let stokenetCandy: ResourceAddress =
+				newResourceAddressPlaceholderStokenetCandy()
+		}
+
+		/// Preview values for `ResourceAddress`, e.g.:
+		/// `ResourceAddress.preview.mainnetXRD`
+		/// or
+		/// `ResourceAddress.preview.stokenetCandy`
+		public static let preview = Preview.of
+	}
+#endif
+
+#if DEBUG
+	extension ResourceAddress: CaseIterable {
+		public typealias AllCases = [Self]
+		public static var allCases: AllCases {
+			let of = Preview.of
+			return [
+				of.mainnetXRD,
+				of.mainnetCandy,
+				of.mainnetNonFungbleGCMembership,
+				of.stokenetXRD,
+				of.stokenetGum,
+				of.stokenetGC,
+				of.stokenetCandy,
+			]
+		}
+	}
+#endif
+
+#if DEBUG
+	/// TODO: Declare in Rust land? Make non-DEBUG?
+	public enum Address: Hashable, Equatable, Sendable {
```

Comment:
This will most probably be needed in the iOS Wallet, where we do need to merge together different kinds of addresses in some places.

Reply:
Will be added in Rust land immediately when I see we really need it. I still - naive perhaps - think we can avoid having it in wallets. Or at least a smaller tagged union (fewer cases)